### ISSUE TITLE
create a generic module for statefulset

### ISSUE THREAD

**Issue by ayushin** [ballj/terraform-kubernetes-postgresql#3]
This is really awesome job.

Is there any plan to create `terraform-kubernetes-statefulset` similar to `https://github.com/ballj/terraform-kubernetes-deployment` to have shared generic module for things like postgresql, redis, mariadb etc

I could use this module for odoo deployment

**Comment by ballj** [comment]
Yeah I plan to do that - I was originally not sure about doing it as I only needed a DB module.

Over the past year or so, I have needed to make the 3 different ones - so a lot of code that could have been reused and has to be maintained separately at the moment.

Should be possible to create, while retaining the possibility of adding DB clustering etc in the future.

Will give it a shot on the weekend, let me know if you have any suggestions.

**Comment by ayushin** [comment]
What would be great is to keep the input variables interface compatible with the deployment as much as possible. As soon as you have it ready I'll try to use it for deploying Odoo instance on our cluster so you'll have some direct feedback.
